src/components/DeleteModal/DeleteModal.tsx

```diff
@@ -0,0 +1,116 @@
+import { toast } from "react-toastify";
+import { deleteData } from "../../core/api/functions.ts";
+import Button from "../Button/Button.tsx";
+import ButtonGrpWrapper from "../Button/buttonGrpWrapper.ts";
+import DeleteModalWrapper from "./../DeleteModal/deleteModal.ts";
+import {
+  DELETE_MODAL_HEADING,
+  CONFIRM_DELETE_TEXT,
+  WARNING_HEADING,
+  WARNING_TEXT,
+} from "./constants/constants.ts";
+import { useLocation, useNavigate, useSearchParams } from "react-router-dom";
+import { useState } from "react";
+import { useAppSelector } from "../../hooks/reduxHooks.ts";
+import { getUrlType } from "../../utils/helper.ts";
+import {
+  H2Styles,
+  LabelStyles,
+  ParagraphStyles,
+} from "../../core/constants/components/text/textStyledComponents.ts";
+
+function DeleteModal({
+  changeDltModalOpenStatus,
+  idArrayToDlt,
+}: {
+  changeDltModalOpenStatus: () => void;
+  idArrayToDlt: string[];
+}) {
+  const [searchParams, setSearchParams] = useSearchParams({ page: "1" });
+  const [confirmDeleteLoader, setConfirmDeleteLoader] = useState(false);
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const confirmDlt = async () => {
+    setConfirmDeleteLoader(true);
+    try {
+      // Use Promise.all to delete all employees concurrently
+      await Promise.all(
+        idArrayToDlt.map(async (employeeId) => {
+          const url = `/employee/${employeeId}`;
+          await deleteData(url); // deleting employee in firebase
+        })
+      );
+
+      if (getUrlType(location.pathname) === "view-employee") navigate("/");
```

Comment:
I think this is not a good idea to relay on url path name, can we use some other way